pkg/blobplugin/main.go

```diff
@@ -83,8 +84,41 @@ func handle() {
 		klog.Fatalf("failed to get Azure Cloud Provider, error: %v", err)
 	}
 	klog.V(2).Infof("cloud: %s, location: %s, rg: %s, VnetName: %s, VnetResourceGroup: %s, SubnetName: %s", cloud.Cloud, cloud.Location, cloud.ResourceGroup, cloud.VnetName, cloud.VnetResourceGroup, cloud.SubnetName)
-
-	driver := blob.NewDriver(&driverOptions, kubeClient, cloud)
+	authProvider, err := azclient.NewAuthProvider(&cloud.ARMClientConfig, &cloud.AzureAuthConfig.AzureAuthConfig)
+	if err != nil {
+		klog.Fatalf("failed to create auth provider: %v", err)
+	}
+	multiTenantAuth, err := authProvider.GetMultiTenantIdentity()
+	if err != nil {
+		klog.Warningf("failed to create multi tenant auth: %v", err)
+	}
+	cred := multiTenantAuth
+	if cred == nil {
+		cred, err = authProvider.GetAzIdentity()
+		if err != nil {
+			klog.Fatalf("failed to create compute auth: %v", err)
+		}
+	}
+	clientFactory, err := azclient.NewClientFactory(&azclient.ClientFactoryConfig{
+		SubscriptionID: cloud.SubscriptionID,
+	}, &cloud.ARMClientConfig, cred)
+	if err != nil {
+		klog.Fatalf("failed to create client factory: %v", err)
+	}
+	networkAuth, err := authProvider.GetNetworkAzIdentity()
+	if err != nil {
+		klog.Warningf("failed to create network auth: %v", err)
```

Comment:
should be Fatal error here?

Reply:
it will never reach here. I'll update authProvider.